src/regexp.c

```diff
@@ -2542,6 +2589,12 @@ static void rex_rec(const unsigned char *code,
         case RE_WB:
             if (!is_word_boundary(ctx, input)) return;
```

Comment:
Maybe I should delete is_word_boundary and use the two new functions? There is some overhead with input validation, but I'm not so sure if it really matters so I left is_word_boundary alone.

Reply:
My hunch is that, since there're lots of dupes in is_begin_of_word_boundary and is_end_of_word_boundary, how about having is_word_boundary(ctx, input, flag) and the flag can switch beginning/end/both?

Reply:
Good idea.